litestar/contrib/pydantic/utils.py

```diff
@@ -158,7 +158,7 @@ def pydantic_get_unwrapped_annotation_and_type_hints(annotation: Any) -> tuple[A
 def is_pydantic_2_model(
     obj: type[pydantic_v1.BaseModel | pydantic_v2.BaseModel],  # pyright: ignore
 ) -> TypeGuard[pydantic_v2.BaseModel]:  # pyright: ignore
-    return issubclass(obj, pydantic_v2.BaseModel)  # pyright: ignore
+    return False if pydantic_v2 is Empty else issubclass(obj, pydantic_v2.BaseModel)  # pyright: ignore
```

Comment:
```suggestion
    return pydantic_v2 is not Empty and issubclass(obj, pydantic_v2.BaseModel)  # pyright: ignore
```

Reply:
I've committed this manually, I couldn't accept it after I had changed the type comment